database/misc/mongodb_replication.py

```diff
@@ -0,0 +1,410 @@
+
+def add_host(module, client, host_name, host_port, host_type, timeout=180, **kwargs):
+    while True:
+        try:
+            admin_db = client['admin']
+            local_db = client['local']
+
+            if local_db.system.replset.count() > 1:
+                module.fail_json(msg='local.system.replset has unexpected contents')
+
+            cfg = local_db.system.replset.find_one()
+            if not cfg:
+                module.fail_json(msg='no config object retrievable from local.system.replset')
+
+            cfg['version'] += 1
+            max_id = max(cfg['members'], key=lambda x:x['_id'])
+            new_host = { '_id': max_id['_id'] + 1, 'host': "{0}:{1}".format(host_name, host_port) }
+            if host_type == 'arbiter':
+                new_host['arbiterOnly'] = True
+
+            if not kwargs['build_indexes']:
+                new_host['buildIndexes'] = False
+
+            if kwargs['hidden']:
+                new_host['hidden'] = True
+
+            if kwargs['priority'] != 1.0:
+                new_host['priority'] = kwargs['priority']
+
+            if kwargs['slave_delay'] != 0:
+                new_host['slaveDelay'] = kwargs['slave_delay']
+
+            if kwargs['votes'] != 1:
+                new_host['votes'] = kwargs['votes']
+
+            cfg['members'].append(new_host)
+            admin_db.command('replSetReconfig', cfg)
+            return
+        except (OperationFailure, AutoReconnect), e:
+            timeout = timeout - 5
+            if timeout <= 0:
+                module.fail_json(msg='reached timeout while waiting for rs.reconfig(): %s' % str(e))
+            time.sleep(5)
+
+def remove_host(module, client, host_name, timeout=180):
+    while True:
+        try:
+            admin_db = client['admin']
+            local_db = client['local']
+
+            if local_db.system.replset.count() > 1:
+                module.fail_json(msg='local.system.replset has unexpected contents')
+
+            cfg = local_db.system.replset.find_one()
+            if not cfg:
+                module.fail_json(msg='no config object retrievable from local.system.replset')
+
+            cfg['version'] += 1
+
+            if len(cfg['members']) == 1:
+                module.fail_json(msg="You can't delete last member of replica set")
+            for member in cfg['members']:
+                if host_name in member['host']:
+                    cfg['members'].remove(member)
+                else:
+                    fail_msg = "couldn't find member with hostname: {0} in replica set members list".format(host_name)
+                    module.fail_json(msg=fail_msg)
+        except (OperationFailure, AutoReconnect), e:
+            timeout = timeout - 5
+            if timeout <= 0:
+                module.fail_json(msg='reached timeout while waiting for rs.reconfig(): %s' % str(e))
+            time.sleep(5)
+
+def load_mongocnf():
```

Comment:
It's just a one of possible solution to store user/password information. You can use this file or login_user/login_password params.